### PR TITLE
spdx to disclosure document: tests

### PR DESCRIPTION
**Pull request by bs-jokri**
PR contains tests for SPDX document parser used for the licenseInfo service. 
It also changes the SPDX parser behavior so that it takes into account the attachment type when deciding whether to parse a file.

### CODE REVIEW

**Comment by alex-evo** [comment]
@maximilianhuber, can you please rebase this onto the current master? I'd like to ask @adrian-evo to test this PR before taking on #206.

File: backend/src/src-licenseinfo/src/test/java/com/siemens/sw360/licenseinfo/parsers/SPDXParserTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                                try {
+                                    attachmentContentStore.put(filename, "");
+                                } catch (SW360Exception e) {
+                                    throw new UncheckedSW360Exception(e);
+                                }
+                                Attachment attachment = makeAttachment(filename, attachmentType);
+                                try {
+                                    assertThat(parser.isApplicableTo(attachment), is(true));
+                                } catch (TException e) {
+                                    e.printStackTrace();
```
Comment (alex-evo) [inline backend/src/src-licenseinfo/src/test/java/com/siemens/sw360/licenseinfo/parsers/SPDXParserTest.java]: I wonder why `SW360Exception` in line 87 is handled nicely, but this one is just `printStackTrace`'d (if you pardon the term)?

Comment (maximilianhuber) [inline backend/src/src-licenseinfo/src/test/java/com/siemens/sw360/licenseinfo/parsers/SPDXParserTest.java]: Yes, you are right, this looks like an autogenerated catch, which was not corrected.

**Comment by alex-evo** [comment]
reviewed-by: alex.borodin@evosoft.com

**Comment by maximilianhuber** [comment]
@alex-evo: The PR should be rebased.

**Comment by alex-evo** [comment]
thanks

**Comment by adrian-evo** [comment]
When trying to download license information with any of the spdx sample files found under spdx/tools/tree/master/Examples, an empty page is displayed in browser and the download file dialog is not displayed. For cli sample files it is working as expected.

**Comment by maximilianhuber** [comment]
I will have a look.

**Comment by maximilianhuber** [comment]
For tests I have used the files found at https://github.com/spdx/tools/tree/master/TestFiles. There are more exmaples which contain yours. **Note:** the current implementation is only able to read `.rdf`-files.

--

I have retested this branch:
1. created project via import from mock
2. added the `SPDXRdfExample-v2.0.rdf` file to an release as `Scan result report (XML)`
3. clicked `Download License Information`

and the returned report contained the expected licenses and copyright information.

**Comment by adrian-evo** [comment]
First issue reported under https://github.com/sw360/sw360portal/issues/217

There is another issue with spdx parser, not sure if this is the intended behavior. The returned report contain only licenses found under extractedText tag, but those from under licenseText and noticeText tags are not extracted. Due to this, from Example-v2.0.rdf are not listed such licenses as  Creative Commons, GNU, Apache Commons Lang, Apache License or other copyright notices.

**Comment by maximilianhuber** [comment]
I will have a look

**Comment by maximilianhuber** [comment]
The PR #229 will tempts to solve the second reported issue of the comment https://github.com/sw360/sw360portal/pull/207#issuecomment-243743512.

**Pull request merged**